packages/contracts-bedrock/src/Safe/VetoerSet/OwnerGuard.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { FixedPointMathLib } from "solady/utils/FixedPointMathLib.sol";
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard } from "safe-contracts/base/GuardManager.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+import { ISemver } from "src/universal/ISemver.sol";
+
+/// @title OwnerGuard
+/// @notice This Guard contract is used to enforce a maximum number of owners and a required
+///         threshold for the Safe Account.
+contract OwnerGuard is ISemver, BaseGuard {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice The initial max owner count used at deployment.
+    uint8 public constant INITIAL_MAX_OWNER_COUNT = 7;
+
+    /// @notice The Safe Account for which this contract will be the guard.
+    Safe public immutable safe;
+
+    /// @notice The maximum number of owners. Can be changed by supermajority.
+    uint8 public maxOwnerCount;
```

Comment:
do we gain anything by having uint8 here? If just deploy cost, I'd say not worth it.

Reply:
No we don't, we could definitely change that.